src/Store/Features.re

```diff
@@ -71,9 +71,21 @@ let update =
     (state, eff);
 
   | Syntax(msg) =>
-    let syntaxHighlights = Feature_Syntax.update(state.syntaxHighlights, msg);
+    let (syntaxHighlights, out) =
+      Feature_Syntax.update(state.syntaxHighlights, msg);
     let state = {...state, syntaxHighlights};
-    (state, Effect.none);
+
+    let effect =
+      switch (out) {
+      | Nothing => Effect.none
+      | ServerError(msg) =>
+        Feature_Notification.Effects.create(
+          ~kind=Error,
+          "Syntax Server error: " ++ msg,
+        )
+        |> Isolinear.Effect.map(msg => Actions.Notification(msg))
+      };
```

Comment:
We should have a helper function for this, internal to this module, as this is going be repeated many, many times here. Might as well add it now.